test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,194 @@
+      return std::move(async_client_);
+    }));
+
+    EXPECT_CALL(factory_context_.cluster_manager_.async_client_manager_,
+                factoryForGrpcService(_, _, _))
+        .WillOnce(Invoke([&](const envoy::config::core::v3::GrpcService&, Stats::Scope&, bool) {
+          return std::move(async_client_factory_);
+        }));
+  }
+}
+void UberFilterFuzzer::fuzzerSetup() {
+  // Setup process when this fuzzer object is constructed.
+  // For a static fuzzer, this will only be executed once.
+
+  // Get the pointer of read_filter when the read_filter is being added to connection_.
+  read_filter_callbacks_ = std::make_shared<NiceMock<Network::MockReadFilterCallbacks>>();
+  ON_CALL(read_filter_callbacks_->connection_, addReadFilter(_))
+      .WillByDefault(Invoke([&](Network::ReadFilterSharedPtr read_filter) -> void {
+        read_filter_ = read_filter;
+        read_filter_->initializeReadFilterCallbacks(*read_filter_callbacks_);
+      }));
+  // Prepare sni for sni_cluster filter and sni_dynamic_forward_proxy filter.
+  ON_CALL(read_filter_callbacks_->connection_, requestedServerName())
+      .WillByDefault(testing::Return("fake_cluster"));
+  // Prepare time source for filters such as local_ratelimit filter.
+  factory_context_.prepareSimulatedSystemTime();
+  // Prepare address for filters such as ext_authz filter.
+  addr_ = std::make_shared<Network::Address::PipeInstance>("/test/test.sock");
+  read_filter_callbacks_->connection_.remote_address_ = addr_;
+  read_filter_callbacks_->connection_.local_address_ = addr_;
+  async_request_ = std::make_unique<Grpc::MockAsyncRequest>();
+}
+
+UberFilterFuzzer::UberFilterFuzzer() : time_source_(factory_context_.simulatedTimeSystem()) {
+  fuzzerSetup();
+}
+bool UberFilterFuzzer::invalidInputForFuzzer(const std::string& filter_name,
+                                             Protobuf::Message* config_message) {
+  // System calls such as reading files are prohibited in this fuzzer. Some input that crashes the
+  // mock/fake objects are also prohibited.
+  const std::string name = Extensions::NetworkFilters::Common::FilterNameUtil::canonicalFilterName(
+      std::string(filter_name));
+  if (filter_name == NetworkFilterNames::get().DirectResponse) {
+    envoy::extensions::filters::network::direct_response::v3::Config& config =
+        dynamic_cast<envoy::extensions::filters::network::direct_response::v3::Config&>(
+            *config_message);
+    if (config.response().specifier_case() ==
+        envoy::config::core::v3::DataSource::SpecifierCase::kFilename) {
+      return true;
+    }
+  } else if (filter_name == NetworkFilterNames::get().LocalRateLimit) {
+    envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit& config =
+        dynamic_cast<envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit&>(
+            *config_message);
+    if (config.token_bucket().fill_interval().seconds() > seconds_in_one_day_) {
```

Comment:
> Should this be an API constraint?

This happens in `SimulatedSystemTime`(when advanceTimeAsync()). I'm not sure whether this will happen in production environment. So I tried to avoid modifying the API.
(In fact there is a constraint: https://github.com/envoyproxy/protoc-gen-validate/blob/3bd820adb86cc455b16b37465905a9b99e22cbbb/templates/cc/duration.go#L12 But I didn't see where `kDurationMaxSeconds` is.)

Reply:
It looks like that's referenced here as a maximum integer, but not as a maximum number of seconds in a day: https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.util.time_util

Reply:
Hmmm I see. I think this may be worth mentioning on the Duration issue on PGV, but let's keep as is and add a comment above the `seconds_in_one_day_` member variable that explain it's limitted to prevent overflow in std::chrono::time_point